src/main/scala/io/getclump/ClumpSource.scala

```diff
@@ -3,6 +3,39 @@ package io.getclump
 import com.twitter.util.Future
 import scala.collection.generic.CanBuildFrom
 
+class ClumpSourceWithParam[T, U, P] private[ClumpSourceWithParam] (val unparameterized: (P) => ClumpSource[T, U]) {
```

Comment:
I think it might be better to name this the same as `ClumpSource` but put under param package so

`io.getclump.paramaterized.ClumpSource`